feature/bgp/gracefulrestart/otg_tests/bgp_graceful_restart_test/bgp_graceful_restart_test.go

```diff
@@ -0,0 +1,677 @@
+		SetChoice(gosnappi.ActionProtocolBgpChoice.INITIATE_GRACEFUL_RESTART).InitiateGracefulRestart().
+		SetPeerNames(peerNames).SetRestartDelay(restartDelay)
+	return grAction
+}
+
+// findProcessByName uses telemetry to find out the PID of a process
+func findProcessByName(ctx context.Context, t *testing.T, dut *ondatra.DUTDevice, pName string) uint64 {
+	pList := gnmi.GetAll(t, dut, gnmi.OC().System().ProcessAny().State())
+	var pID uint64
+	for _, proc := range pList {
+		if proc.GetName() == pName {
+			pID = proc.GetPid()
+			t.Logf("Pid of daemon '%s' is '%d'", pName, pID)
+		}
+	}
+	return pID
+}
+
+// gNOIKillProcess kills a daemon on the DUT, given its name and pid.
+func gNOIKillProcess(ctx context.Context, t *testing.T, dut *ondatra.DUTDevice, pName string, pID uint32) {
+	gnoiClient := dut.RawAPIs().GNOI(t)
+	killRequest := &gnps.KillProcessRequest{Name: pName, Pid: pID, Signal: gnps.KillProcessRequest_SIGNAL_TERM, Restart: true}
+	killResponse, err := gnoiClient.System().KillProcess(context.Background(), killRequest)
+	t.Logf("Got kill process response: %v\n\n", killResponse)
+	if err != nil {
+		t.Fatalf("Failed to execute gNOI Kill Process, error received: %v", err)
+	}
+}
+
+func TestTrafficWithGracefulRestartReceiver(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	ate := ondatra.ATE(t, "ate")
+
+	// Configure interface on the DUT
+	t.Log("Start DUT interface Config")
+	configureDUT(t, dut)
+
+	// Configure BGP+Neighbors on the DUT
+	t.Log("Configure BGP with Graceful Restart option under Global Bgp")
+	dutConfPath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP")
+	nbrList := buildNbrList(ateAS)
+	dutConf := bgpWithNbr(dutAS, nbrList, dut)
+	gnmi.Replace(t, dut, dutConfPath.Config(), dutConf)
+	fptest.LogQuery(t, "DUT BGP Config", dutConfPath.Config(), gnmi.GetConfig(t, dut, dutConfPath.Config()))
+
+	// ATE Configuration.
+	t.Log("Start ATE Config")
+	config := configureATE(t, ate)
+
+	// Verify Port Status
+	t.Run("verifyDUTPorts", func(t *testing.T) {
+		t.Log("Verifying port status")
+		verifyPortsUp(t, dut.Device)
+	})
+	t.Run("VerifyBGPParameters", func(t *testing.T) {
+		t.Log("Check BGP parameters")
+		checkBgpStatus(t, dut)
+	})
+
+	var startTime time.Time
+	// Send Graceful Restart Trigger from ATE to DUT within the GR timer configured on the DUT
+	t.Run("InitiateGracefulRestartWithinDUTGRTimer", func(t *testing.T) {
+		// starting traffic before graceful restart
+		t.Log("Starting traffic before Graceful restrat Trigger from OTG")
+		ate.OTG().StartTraffic(t)
+		startTime = time.Now()
+		t.Log("Send Traffic while GR timer counting down. Traffic should pass as BGP GR is enabled!")
+		t.Log("Send Graceful Restart Trigger from OTG to DUT -- trigger GR Restart timer(60s) < configured GR restart Timer(120s)")
+		ate.OTG().SetControlAction(t, createGracefulRestartAction(t, []string{"atedst.BGP4.peer"}, 60))
```

Comment:
Move 60 to a constant next to grTime so that it's easier to see when both should be changed.

Reply:
Done